src/ggl-setup.py

```diff
@@ -0,0 +1,524 @@
+            "liburiparser-dev", "cgroup-tools"
+        ]
+
+        # Update package list
+        update_command = ["sudo", "apt", "update"]
+        subprocess.run(update_command, check=True)
+
+        # Install packages
+        install_command = ["sudo", "apt", "install", "-y"] + packages
+        subprocess.run(install_command, check=True)
+
+        print("Successfully updated and installed dependencies")
+        return True
+
+    except Exception as e:
+        print(f"Error when installing dependencies: {e}", file=sys.stderr)
+        return False
+
+
+def _add_user_and_group(user: str, group: str):
+    try:
+        # Create the group
+        if not _check_group_exists(group):
+            subprocess.run(['sudo', 'groupadd', group], check=True)
+            print(f"Successfully created {group} group")
+
+        # Create the user and add to the group
+        if not _check_user_exists(user):
+            subprocess.run(['sudo', 'useradd', '-Ng', group, user], check=True)
+            print(f"Successfully created {user} user")
+        return True
+
+    except subprocess.CalledProcessError as e:
+        print(f"Command failed: {e.cmd}. Return code: {e.returncode}")
+        return False
+    except Exception as e:
+        print(f"Error when adding user and group: {str(e)}")
+        return False
+
+
+def _delete_user_and_group(user: str, group: str):
+    try:
+        # Delete the user
+        if _check_user_exists(user):
+            subprocess.run(['sudo', 'userdel', user], check=True)
+            print(f"Successfully deleted {user} user")
+
+        # Delete the group
+        if _check_group_exists(group):
+            subprocess.run(['sudo', 'groupdel', group], check=True)
+            print(f"Successfully deleted {group} group")
+        return True
+
+    except subprocess.CalledProcessError as e:
+        print(f"Command failed: {e.cmd}. Return code: {e.returncode}")
+        return False
+    except Exception as e:
+        print(f"Error when deleting user and group: {str(e)}")
+        return False
+
+
+def _check_group_exists(group: str) -> bool:
+    try:
+        grp.getgrnam(group)
+        return True
+    except KeyError:
+        return False
+
+
+def _check_user_exists(user: str) -> bool:
+    try:
+        pwd.getpwnam(user)
+        return True
+    except KeyError:
+        return False
+
+
+def _build_with_cmake() -> bool:
+    original_dir = os.getcwd()
+    try:
+        # CMake configure command
+        cmake_cmd = [
+            'cmake', '-B', 'build', '-D', 'CMAKE_INSTALL_PREFIX=/usr/local',
+            '-D', 'CMAKE_BUILD_TYPE=MinSizeRel', '-D', 'CMAKE_BUILD_TYPE=Debug',
+            '-DGGL_LOG_LEVEL=DEBUG'
+        ]
+
+        subprocess.run(cmake_cmd, check=True)
+
+        # Change the directory to build
+        os.chdir("build")
+        build_cmd = ['make', '-C', 'build', '-j$(nproc)']
+        subprocess.run(build_cmd, shell=True, check=True)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

subprocess call with shell=True identified, security issue. https://bandit.readthedocs.io/en/latest/plugins/b602_subprocess_popen_with_shell_equals_true.html